src/main/java/Deadline.java

```diff
@@ -0,0 +1,49 @@
+import java.time.LocalDateTime;
+import java.time.DateTimeException;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+
+public class Deadline extends Task {
+    protected LocalDateTime byDateTime;
+
+    public Deadline(String description, String byDateTime) throws DukeException {
+        super(description);
+        this.taskType = TaskType.DEADLINE;
+        try {
+            this.byDateTime = LocalDateTime.parse(byDateTime, DateTimeFormatter.ofPattern("dd/MM/yyyy HH:mm"));

```

Comment:
Thanks for telling me about this tip! Didn't realise there is an easier way to format date and time.